### DISCUSSION TITLE
What is the point of both `build` and `build_any`?

### DISCUSSION THREAD

**Discussion by tqwewe** [SeaQL/sea-query#139] [Q&A] [answered]
I see sea-query has two common functions throughout traits: `build()` and `build_any()`

What is the purpose of having both of these? I see that the source code for it is almost completely identical if I'm not mistaken.
The only difference I notice, is that `build_any` takes `dyn`, where `build` takes `impl`.

Are both these functions necessary?

**Comment by tyt2y3** [comment, marked as answer]
`build` is static dispatch, while `build_any` is dynamic dispatch. So if the user only ever uses one database variant, they'd simply call `build`.
The `any` is only useful for this case where the `QueryBuilderBackend` can be swapped runtime.
But even SeaORM uses a enum and match statement, so I guess 'in the wild usage' is rare.
